src/Microsoft.DocAsCode.Metadata.ManagedReference.Roslyn/Visitors/CSYamlModelGenerator.cs

```diff
@@ -935,95 +931,72 @@ where object.Equals(value, pair.ConstantValue)
-                        var cv = (int)constantValue;
-                        if (cv == 0)
-                        {
-                            return v == 0;
-                        }
-                        return (v & cv) == cv;
-                    }
+                    return GetFlagExpressions(flags.Select(p => (p.Name, (int)p.ConstantValue)), (int)value, namedType);
                 case SpecialType.System_UInt32:
-                    {
-                        var v = (uint)value;
-                        var cv = (uint)constantValue;
-                        if (cv == 0)
-                        {
-                            return v == 0;
-                        }
-                        return (v & cv) == cv;
-                    }
+                    return GetFlagExpressions(flags.Select(p => (p.Name, (uint)p.ConstantValue)), (uint)value, namedType);
                 case SpecialType.System_Int64:
-                    {
-                        var v = (long)value;
-                        var cv = (long)constantValue;
-                        if (cv == 0)
-                        {
-                            return v == 0;
-                        }
-                        return (v & cv) == cv;
-                    }
+                    return GetFlagExpressions(flags.Select(p => (p.Name, (long)p.ConstantValue)), (long)value, namedType);
                 case SpecialType.System_UInt64:
-                    {
-                        var v = (ulong)value;
-                        var cv = (ulong)constantValue;
-                        if (cv == 0)
-                        {
-                            return v == 0;
-                        }
-                        return (v & cv) == cv;
-                    }
+                    return GetFlagExpressions(flags.Select(p => (p.Name, (ulong)p.ConstantValue)), (ulong)value, namedType);
                 default:
-                    return false;
+                    return Array.Empty<ExpressionSyntax>();
             }
         }
 
+        private static IEnumerable<ExpressionSyntax> GetFlagExpressions<T>(IEnumerable<(string Name, T Value)> flags, T value, INamedTypeSymbol namedType) where T : unmanaged
+        {
+            var enumType = GetTypeSyntax(namedType);
+            if (EqualityComparer<T>.Default.Equals(value, default))
+            {
+                string defaultFlagName = flags.FirstOrDefault(f => EqualityComparer<T>.Default.Equals(f.Value, default)).Name;
+                return defaultFlagName != null ? new[] { GetFlagExpression(defaultFlagName) } : Array.Empty<ExpressionSyntax>();
+            }
+            var negativeFlags = flags.Where(p => Comparer<T>.Default.Compare(p.Value, default) < 0);
+            var positiveFlags = flags.Where(p => Comparer<T>.Default.Compare(p.Value, default) > 0);
+            var sortedFlags = negativeFlags.OrderByDescending(p => p.Value).Concat(positiveFlags.OrderByDescending(p => p.Value)).ToList();
```

Comment:
Is `sortedFlags` sorted by bit count? If so, `0b_1000` > `0b_0111` but has less bit count.

Reply:
It's sorted by descending value, not by bit count. I've yet to run into a situation where this is an issue in practice - unless you have something **very** odd going on with your bit flags, it works out fine because all that matters is that flags that hold multiple values are considered before the individual values, which this accomplishes.

Reply:
This does not affect the output because you would match the higher bits first.